lib/src/mock.dart

```diff
@@ -499,17 +503,21 @@ class PostExpectation<T> {
   /// Note: [expected] cannot be a Future or Stream, due to Zone considerations.
   /// To return a Future or Stream from a method stub, use [thenAnswer].
   void thenReturn(T expected) {
-    if (expected is Future) {
-      throw ArgumentError('`thenReturn` should not be used to return a Future. '
-          'Instead, use `thenAnswer((_) => future)`.');
-    }
-    if (expected is Stream) {
-      throw ArgumentError('`thenReturn` should not be used to return a Stream. '
-          'Instead, use `thenAnswer((_) => stream)`.');
-    }
+    _throwIfInvalid(expected);
     return _completeWhen((_) => expected);
   }
 
+  void thenReturnInOrder(List<T> expects) {
+    if (expects.isEmpty) {
+      throw ArgumentError('thenReturnInOrder expects should not be empty');
+    }
+
+    expects.forEach(_throwIfInvalid);
+
+    final answers = Queue.of(expects);
+    thenAnswer((_) => answers.removeFirst());
```

Comment:
Should we check `answers` inside the closure and report a more specific error than "`Queue.removeFirst()` called on empty queue" if a method was called more times than the length of the given list?

Reply:
Thanks for the review, I made an update with your suggestion, do you see that the thrown error is fine?